tools/cr/plaster_test.py

```diff
@@ -255,6 +255,178 @@ def __init__(self):
             output = stderr.getvalue()
             self.assertIn(str(changed_path), output)
 
+    def test_pattern_validation_failures(self):
+        """Test various pattern validation failures."""
+        test_file_chromium = Path(
+            'chrome/common/extensions/api/test_validation.idl')
+
+        # Write and commit file to chromium repository
+        self.fake_chromium_src.write_and_stage_file(
+            test_file_chromium, 'Initial content for Chromium file.',
+            self.fake_chromium_src.chromium)
+        self.fake_chromium_src.commit('Add test_validation.idl',
+                                      self.fake_chromium_src.chromium)
+
+        # Test various invalid validation cases
+        validation_cases = [
+            # (toml_content, expected_error_message)
+            ('''
+              [[substitution]]
+              description = 'Both patterns specified'
+              pattern = 'Chromium'
+              re_pattern = 'Chromium'
+              replace = 'Plaster'
+            ''', 'Please specify either pattern or re_pattern'),
+            ('''
+              [[substitution]]
+              description = 'No pattern specified'
+              replace = 'Plaster'
+            ''', 'No pattern specified'),
+            ('''
+              [[substitution]]
+              description = 'No replace specified'
+              pattern = 'Chromium'
+            ''', 'No replace value specified'),
+        ]
+
+        for toml_content, expected_error in validation_cases:
+            with self.subTest(error=expected_error):
+                # Create a PlasterFile with invalid configuration
+                plaster_path = plaster.PLASTER_FILES_PATH / (
+                    str(test_file_chromium) + '.toml')
+                plaster_path.parent.mkdir(parents=True, exist_ok=True)
+                plaster_path.write_text(toml_content)
+
+                plaster_file = plaster.PlasterFile(plaster_path)
+                with self.assertRaises(ValueError) as context:
+                    plaster_file.apply()
+
+                self.assertIn(expected_error, str(context.exception))
+
+    def test_pattern_exact_match_works(self):
```

Comment:
This test and the ones below seem to have the same code pattern. Maybe make it into a function that takes the test plaster file content and expected result as parameters?